openmls/src/extensions/required_capabilities.rs

```diff
@@ -31,37 +40,49 @@ use super::{Deserialize, ExtensionError, ExtensionType, Serialize};
     TlsSize,
 )]
 pub struct RequiredCapabilitiesExtension {
-    extensions: Vec<ExtensionType>,
-    proposals: Vec<ProposalType>,
+    extension_types: Vec<ExtensionType>,
+    proposal_types: Vec<ProposalType>,
+    credential_types: Vec<CredentialType>,
 }
 
 impl RequiredCapabilitiesExtension {
     /// Creates a new [`RequiredCapabilitiesExtension`] from extension and proposal types.
-    pub fn new(extensions: &[ExtensionType], proposals: &[ProposalType]) -> Self {
+    pub fn new(
+        extension_types: &[ExtensionType],
```

Comment:
Should this take `Vec`s while we're here?

Reply:
Yeah. This is very common in OpenMLS. Maybe we should define a tracking issue "Revisit Rust API Guidelines" or so and add `TODO(C-CALLER-CONTROL)` whenever we see it?